mlflow/server/handlers.py

```diff
@@ -1599,18 +1596,6 @@ def _validate_source(source: str, run_id: str) -> None:
             INVALID_PARAMETER_VALUE,
         )
 
-    # There might be file URIs that are local but can bypass the above check. To prevent this, we
-    # disallow using file URIs as model version sources by default unless it's explicitly allowed
-    # by setting the MLFLOW_ALLOW_FILE_URI_AS_MODEL_VERSION_SOURCE environment variable to True.
-    if not MLFLOW_ALLOW_FILE_URI_AS_MODEL_VERSION_SOURCE.get() and is_file_uri(source):
```

Comment:
Now that is_local_uri always returns True if is_local_file is True, so we don't need this check as it will be captured by above `if is_local_uri(source):` check

Reply:
To confirm, we never reach if `source` is a file URI?

Reply:
Yes, if source is a file URI, it would go to `if is_local_uri(source): ` block